src/uct/ib/mlx5/ib_mlx5.inl

```diff
@@ -175,6 +175,40 @@ uct_ib_mlx5_inline_copy(void *restrict dest, const void *restrict src, unsigned
 }
 
 
+/**
+ * Copy uct_iov_t array to inline segment, taking into account QP wrap-around.
+ *
+ * @param dest     Inline data in the WQE to copy to.
+ * @param iov      A pointer to an array of uct_iov_t elements.
+ * @param iov_cnt  A number of elements in a iov array.
+ * @param length   Total data length.
+ * @param wq       Send work-queue.
+ */
+static UCS_F_ALWAYS_INLINE void
+uct_ib_mlx5_inline_iov_copy(void *restrict dest, const uct_iov_t *iov, size_t iovcnt,
+                            size_t length, uct_ib_mlx5_txwq_t *wq)
+{
+    size_t iov_it;
+    ptrdiff_t n;
+
+    ucs_assert(dest != NULL);
+
+    n = UCS_PTR_BYTE_DIFF(dest, wq->qend);
+    if (n >= length) {
+        uct_iov_to_buffer(iov, iovcnt, 0, dest);
+    } else {
+        for (iov_it = 0; iov[iov_it].length < n; ++iov_it) {
+            memcpy(dest, iov[iov_it].buffer, iov[iov_it].length);
+            dest = UCS_PTR_BYTE_OFFSET(dest, iov[iov_it].length);
+            n   -= iov[iov_it].length;
+        }
+        memcpy(dest, iov[iov_it].buffer, n);
```

Comment:
maybe add size limit to uct_iov_to_buffer() copy ("n"), so it could be reused for this loop as well?

Reply:
It can be done. However, the limit will cause additional conditions in uct_iov_to_buffer which will hurt performance. E.g. the loop will have at least one more bound condition: `iov_it < iovcnt && copied <= copy_limit`.

Reply:
i'd assume that if copy_limit is a constant SIZE_MAX, the compiler could eliminate it